lua/neogit/popups/commit/actions.lua

```diff
@@ -105,9 +110,11 @@ local function commit_special(popup, method, opts)
 end
 
 function M.commit(popup)
-  if not git.status.anything_staged() and not allow_empty(popup) then
-    notification.warn("No changes to commit.")
-    return
+  if not config.values.commit_editor.fast then
```

Comment:
We could also modify the `git.status.anything_staged()` implementation to check `git.repo.state.staged.items[1] ~= nil` if "fast" is set to true. Then we don't need to skip this check, but instead just rely on the repo state data already in memory.

Reply:
We could also put `not allow_empty(popup)` _before_ `git.status.anything_changed()`, so you could short-circuit the conditional by enabling "--allow-empty".